include/utils/message_queue.hpp

```diff
@@ -51,10 +51,16 @@ class MessageQueue {
 
 /// \brief Contains a message queue driven list of handler callbacks
 class MessageHandler {
+public:
+    struct MessageDetails {
+        std::string topic;
+        std::shared_ptr<json> data;
```

Comment:
`unique_ptr` should probably be enough here

Reply:
The existing code uses shared_ptr. I can change it but I don't know the original reason for using shared_ptr.

Reply:
Yes, I know - then lets hope it will be refactored in a following commit.

Reply:
Done some testing and it needs to be a shared_ptr.
in `MQTTAbstractionImpl::on_mqtt_message()` the same JSON message could be added to multiple message handlers. There is a vector of handlers and the same JSON data is added to all that match the required topic.
With unique_ptr that JSON data would be copied.